proxy_docker/app/script/call_lightningd.sh

```diff
@@ -327,7 +327,7 @@ ln_pay() {
     # The amount must match if not "any"
     # If the amount is not in the invoice and not supplied as expected_msatoshi, then both will be null, that's ok!
     # Same thing goes for the description.
-    if [ "${expected_msatoshi}" != "${invoice_msatoshi}" ] && [ "${invoice_msatoshi}" != "null" ]; then
+    if [ -n "${expected_msatoshi}" ] && [ "${expected_msatoshi}" != "null" ]  &&  [ "${expected_msatoshi}" != "${invoice_msatoshi}" ] && [ "${invoice_msatoshi}" != "null" ]; then
```

Comment:
This is superfluous.  The previous condition included what you added here.

Reply:
expected_msatoshi not set or something other than string null would pass in previous conditional

Reply:
`expected_msatoshi` will always be set to a value or `null` because of `expected_msatoshi=$(echo "${request}" | jq ".expected_msatoshi")` above.  If `$request` is not parsable by jq, we won't reach this part of the code.

Reply:
Yes i understand but the previous condition does not test for `expected_msatoshi`'s value, only that it's *not equal to `invoice_msatoshi`* and that `invoice_msatoshi` is not null:
   ` if [ "${expected_msatoshi}" != "${invoice_msatoshi}" ] && [ "${invoice_msatoshi}" != "null" ]; then`
so if `expected_msatoshi` is _not set, IE `null`_ , then this condition would evaluate to `true` since `expected_msatoshi != invoice_msatoshi` would evaluate to true (assuming `invoice_msatoshi` *is set*).
Thus the new condition
    `if [ -n "${expected_msatoshi}" ] && [ "${expected_msatoshi}" != "null" ]  &&  [ "${expected_msatoshi}" != "${invoice_msatoshi}" ] && [ "${invoice_msatoshi}" != "null" ]; then`
Checks that indeed `expected_msatoshi` is set and not equal to null and then tests it's value in relation to `invoice_msatoshi`.

Reply:
You are absolutely right!

Simply put, if `invoice_msatoshi` is null (donation invoice), `expected_msatoshi` must not be null/empty but previous cond was accepting it as valid (reach the else block).

I think we should also check that `expected_msatoshi` is > 0 ...  I suspect sending a 0 msat payment would upset c-lightning... What do you think?  Oh well, we would then just return the c-lightning error response, which is fine.